src/AutoLaunchMac.coffee

```diff
@@ -1,33 +1,58 @@
-applescript = require 'applescript'
+fs = require('fs')
+mkdirp = require('mkdirp')
+untildify = require('untildify')
 
-tellTo = 'tell application "System Events" to '
+dir = untildify('~/Library/LaunchAgents')
 
 module.exports =
+    getFile: (opts) ->
+        file = dir+opts.appName+'.plist'
+        return file
 
     enable: (opts) ->
-        new Promise (resolve, reject) ->
-            isHidden = if opts.isHiddenOnLaunch then 'true' else 'false'
-            properties = "{path:\"#{opts.appPath}\", hidden:#{isHidden}, name:\"#{opts.appName}\"}"
-            command = "#{tellTo} make login item at end with properties #{properties}"
-
-            applescript.execString command, (err) ->
+        new Promise (resolve, reject) =>
+            file = @getFile(opts)
+            array = [opts.appPath]
+            if(opts.isHiddenOnLaunch) then array.push('--hidden')
+            command = array.map(a => `    <string>${a}</string>`)
```

Comment:
Wait... backticks in CoffeeScript? Never saw this before. I thought you had to use double quotes for interpolation.